File: 129. Sum Root to Leaf Numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    let ans = 0;
    let helper = function(root, s) {
        if(root == null) return;

        if(root.left == null && root.right == null) {
            ans += s * 10 + root.val;
            return;
        }

        helper(root.left, s * 10 + root.val);
        helper(root.right, s * 10 + root.val);
    }
    
    helper(root, 0);
    return ans;
};

File: 129. Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.ans = 0

    def helper(self, root: Optional[TreeNode], val: int):
        if(root == None): 
            return
        
        if(root.left == None and root.right == None):
            self.ans += root.val + val * 10
            return
        
        self.helper(root.left, val * 10 + root.val)
        self.helper(root.right, val * 10 + root.val)

    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.helper(root, 0)
        return self.ans
